### PR TITLE
Renders title properly in case of multiple controller selection

### LINKED ISSUE #1166: Ensure rendering of multiple controllers with either flex CSS or ellipsis shortening
The user can select as many controllers as they choose and the PageHeaderLayout component will wrap the title appropriately. The title may be useful to confirm the controllers the user selected but we should eventually display a controller tag next to each result selected.

### PR DESCRIPTION
**Pull request by Anisha1234**
Fixes #1166 
- The user can select as many controllers as they choose and the PageHeaderLayout component will wrap the title appropriately. 

- The title may be useful to confirm the controllers the user selected. 

- Displays controller tag next to each result selected.

### CODE REVIEW

File: web-server/v0.4/src/utils/parse.js [hunk 1]
```diff
@@ -12,6 +14,12 @@ export const parseIterationData = results => {
         dataIndex: 'iteration_name',
         width: 150,
         key: 'iteration_name',
+        render: (text, record) => (
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/utils/parse.js]: We should render controller names next to each result name as opposed to the iteration name (we can assume all iterations from a result belong to the same controller)

File: web-server/v0.4/src/components/PageHeader/index.js [hunk 2]
```diff
@@ -204,6 +207,29 @@ export default class PageHeader extends PureComponent {
...
+    let titleHeader = '';
+    if (selectedControllers !== undefined) {
+      titleHeader = selectedControllers.map(i => (
+        <Tag color="blue" style={{ fontSize: '105%' }}>
+          {i}
+        </Tag>
+      ));
+      if (selectedControllers.length > 1) {
+        titleHeader = (
+          <Collapse defaultActiveKey={['1']} style={{ marginBottom: '0.5%' }}>
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.js]: If we plan on using `Tag` to display controller names, we can simply render the tags without the `Collapse` component as the tags will be wrapped in the `PageHeader` component. We can use this approach for `selectedControllers` with a length of 1 or more.

File: web-server/v0.4/src/components/PageHeader/index.js [hunk 3]
```diff
@@ -193,9 +194,11 @@ export default class PageHeader extends PureComponent {
       tabActiveKey,
       tabDefaultActiveKey,
       tabBarExtraContent,
+      selectedControllers,
     } = this.props;
-    const { breadcrumb } = this.state;
 
+    const { breadcrumb } = this.state;
+    // console.log('nfjewq');
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.js]: We have extra `console.log` statements here and on line 117 in `web-server/v0.4/src/pages/ComparisonSelect/index.js`

File: web-server/v0.4/src/components/PageHeader/index.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React, { PureComponent, createElement } from 'react';
 import PropTypes from 'prop-types';
 import pathToRegexp from 'path-to-regexp';
-import { Breadcrumb, Tabs } from 'antd';
+import { Breadcrumb, Tabs, Tag, Col } from 'antd';
```
Comment (portante) [inline web-server/v0.4/src/components/PageHeader/index.js]: Are `Tag` and `Col` predefined to properly display title-case?

File: web-server/v0.4/src/components/PageHeader/index.js [hunk 5]
```diff
@@ -204,6 +205,19 @@ export default class PageHeader extends PureComponent {
...
       activeKeyProps.activeKey = tabActiveKey;
     }
+    let titleHeader = '';
+    if (selectedControllers !== undefined) {
+      titleHeader = selectedControllers.map(i => (
+        <Tag color="blue" style={{ fontSize: '105%' }}>
+          {i}
+        </Tag>
+      ));
+      titleHeader = <p>Selected Controllers : {titleHeader} </p>;
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.js]: Can we consider referencing the titleHeader as `titleHeader = <div><h1 className={styles.title}>Selected Controllers</h1>{titleHeader}</div>` for visual consistency with title headers in other page components?

File: web-server/v0.4/src/components/PageHeader/index.js [hunk 6]
```diff
@@ -204,6 +205,19 @@ export default class PageHeader extends PureComponent {
...
+    }
+
+    if (selectedControllers === undefined) {
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.js]: What about simply referencing this logic as the `else` part of the `if (selectedControllers !== undefined)` logic mentioned above?

File: web-server/v0.4/src/pages/ComparisonSelect/index.js [hunk 7]
```diff
@@ -133,7 +135,15 @@ class ComparisonSelect extends React.Component {
               };
               return (
                 <div key={iteration.resultName} style={{ marginTop: 32 }}>
-                  <h2>{iteration.resultName}</h2>
+                  <h2>
+                    {iteration.resultName}{' '}
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/pages/ComparisonSelect/index.js]: We should use CSS for all styling/formatting as opposed to inserting whitespace into the JSX. I highly recommend CSS Flexbox for the kind of structured spacing we are looking for here: https://www.w3schools.com/css/css3_flexbox.asp

File: web-server/v0.4/src/pages/ComparisonSelect/index.js [hunk 8]
```diff
@@ -133,7 +135,15 @@ class ComparisonSelect extends React.Component {
...
+                    <span>
+                      <Tag color="blue" style={{ fontSize: '70%' }}>
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/pages/ComparisonSelect/index.js]: Is there a specific reason we use percentage for fontSizes? I believe ant.design automatically scales fontSizes for their components and we usually use the default font sizes for all components within the dashboard.

File: web-server/v0.4/src/pages/ComparisonSelect/index.js [hunk 9]
```diff
@@ -133,7 +135,13 @@ class ComparisonSelect extends React.Component {
               };
               return (
                 <div key={iteration.resultName} style={{ marginTop: 32 }}>
-                  <h2>{iteration.resultName}</h2>
+                  <div style={{ display: 'flex' }}>
+                    <h1>{iteration.resultName}</h1>
+                    <span>
+                      <Tag color="blue">{iteration.controllerName}</Tag>
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/pages/ComparisonSelect/index.js]: We need slight margin on each Tag (`marginLeft: 8` should work) for better readability.

**Comment by portante** [comment]
Marking WIP until the various comments are resolved.

File: web-server/v0.4/src/components/PageHeader/index.js (hunk 4 above)
Comment (Anisha1234) [reply web-server/v0.4/src/components/PageHeader/index.js]: These are predefined components of Ant-Design. However we don't need Col anymore.

**Pull request merged**